This page lists the data sets used in the paper. PerM has been tested on many other public and private data sets. For more public data sets, check the [NCBI Short read archive](http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?) and the [Applied Biosystems](http://solidsoftwaretools.com/gf/project/ecoli2x50/) or [Illumina](ftp://ftp.illumina.com) websites.

### Reference Data Sets ###
  * The anthracis.fasta file can be downloaded from [SOCS download](http://socs.biology.gatech.edu/Download.html) page.
  * Both repeat masked and non-masked human genomes can be downloaded from UCSC's genome browser. The file names are chromFaMasked.zip and chromFa.zip: [Human Genome from UCSC](http://hgdownload.cse.ucsc.edu/goldenPath/hg18/bigZips).
### Read files ###
  * Illumina Test data used in the revised paper [SRR001154](http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?cmd=viewer&m=data&s=viewer&run=SRR001154)
  * SOLiD Read Test data used in the paper. [ERR00455](http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?cmd=viewer&m=data&s=viewer&run=ERR000455&search=&x=&y=&show_read=on&show_graph=on&space=color)
The data downloaded from the NCBI SRA is in the base format. We translated the data into
color signals in CSFASTA format. Because the data set is too large for many people to download, we are provide the subset we used in our download section (ERR000455\_5M.tgz).
  * Real SOLiD reads from B. anthracis can be downloaded from [SOCS's download page](http://socs.biology.gatech.edu/Download.html).
  * Simulated SOLiD or Solexa reads can be generated using the small program in the download section.
[Back to manual](http://code.google.com/p/perm/wiki/Manual)