We are building the pipeline with PerM to deal with the RNA-seq data and estimate the expression level of each gene/isoform. 
 # The putative transcripts are enumerated from the region with coverage and  the known and detected splicing junctions.
 # These transcript are filtered by similar scoring scheme adopted from the [http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/ AceView] data base.  
 # The RNA-seq reads are mapped to the enumerated transcripts. 
 # The mapping results are sent to *TranSEQ* program, Dr. Xiting Yan's (unpublished) program for expression levels estimation.

The preliminary [http://dw408k-1.cmb.usc.edu:8080/Gene_List.htm result] can be browsed on the UCSC [http://genome.ucsc.edu/ genome browser], with the following steps. 

 * Follow the hyperlink to check a gene in a chosen chromosome. 
 * Click the {{{Expression}}} link of each gene to see the mapped-reads coverage on the UCSC genome browser. 
 * Switch the reference to hg18 by clicking the {{{Convert}}} button on the top tool bar and choosing the {{{New Assembly}}} to {{{Mar.2006}}} 
 * Go back to gene list page and click the selected {{{Isoforms}}} of the gene to see the transcripts we enumerated. 
 * Copy the isoform sequence and [http://genome.ucsc.edu/cgi-bin/hgBlat?hgsid=148468955 BLAT] it on the genome browser to see it with the mapped reads signal.
 * The expressed proportion and the expressed log likelihood of the transcripts are shown on the isoforms Id. 
 * The log likelihood shows the confidence that the transcript is expressed. Inf log likelihood indicates there are reads uniquely mapped to the isoform.
 * The result can be compared with the [http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/ AceView]'s transcripts by [http://genome.ucsc.edu/cgi-bin/hgBlat?hgsid=148468955 BLAT]ing the result together.
 * The result of nucleotide sequence and the amino sequence should be [http://genome.ucsc.edu/cgi-bin/hgBlat?hgsid=148468955 BLAT] separately.  