We are building the pipeline with PerM to deal with the RNA-seq data and estimate the expression level of each gene/isoform. 
 # The putative transcripts are enumerated with the known and detected splicing junctions, using similar scoring scheme adopted from [http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/ AceView] data base. 
 # The RNA-seq reads are mapped to the enumerated transcripts. 
 # The mapping results are sent to *TranSEQ* program, Dr. Xiting Yan's (unpublished) program for expression level's estimation.

The preliminary [http://dw408k-1.cmb.usc.edu:8080/Gene_List.htm result] can be browsed on the UCSC genome browser. 

 * Follow the hyperlink to check a gene in a chosen chromosome. 
 * Click the _Expression_ link to see the mapped-reads coverage on the UCSC genome browser. 
 * Remember to switch the reference to hg18 by clicking the {{{Convert}}} button on the top tool bar and choosing the {{{New Assembly}}} to {{{Mar.2006}}} 
 * Go back to previous page and click the selected _Isoforms_ of the gene to see the transcripts we enumerated. 
 * Copy the isoform sequence and [http://genome.ucsc.edu/cgi-bin/hgBlat?hgsid=148468955 BLAT] it on the genome browser to see it with the mapped reads signal.
 * On the isoform ID, we attached the expression proportion and the expressed log likelihood. The log likelihood shows the confidence that the transcript is expressed. Inf log likelihood indicates there are reads uniquely mapped to the isoform.
 * The result can be compared with the [http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/ AceView]'s transcripts by [http://genome.ucsc.edu/cgi-bin/hgBlat?hgsid=148468955 BLAT]ing the result together.
 * The result of nucleotide sequence and the amino sequence should be [http://genome.ucsc.edu/cgi-bin/hgBlat?hgsid=148468955 BLAT] separately.  