PerM takes (1) Reference and (2)Reads as the input files.
==Reference Files==
PerM takes single or multiple fasta files as the reference. The file should have a corresponding ext name, such as '.fasta, .fa, .fna or .mfa'. 
PerM takes fasta with multiple sequences separated by a line start with "> Sequence Name", which is typically gene transcripts or contigs. The first word after ">" will be used as a "Sequence Name", which will be showed in the alignments result output. 
For example, in reads.fa: 
{{{
>gene1 
ACGTACGTACGTACGTACGTACGTACGTACGT
>gene2 
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
...
>geneN
GGGGAAAATTTAGCGGGGGAAAATTTAGCGGGGGA
AAATTTAGCG...
}}}
"geneN" will be the Id for the reference sequence and be printed in the mapping result. 
If there are multiple fasta files, make a list file with ext name '.txt', ex: {{{HumanGenome.txt}}}, with each fasta file path in a row. For example: 
{{{
/data/chr1.fa
/data/chr2.fa
...
/data/chrY.fa
}}}
Each file, ex /data/chr1.fa can have one or multiple sequences.
==== Non-ACGT characters ====
Inside our program, sequences are concatenated and separated by "N", so no junction will be aligned. Only ACGT and N are allowed in the reference file. Other valid nucleotide symbol such as U, R, Y, M, K, S, W, B, D, H, V will be modified to "N". 

==== Index file ====
PerM can saved the preprocessed reference into binary index file, by specify the argument.
{{{ -m -s /PATH.index }}}. The default index file name, {{{<read_length>_<Seed>_<Read type>_<RefName>_v2.index}}}, is used

Limited by OS the system, the index file name couldn't be too long to be saved. 

==Read Files==
PerM expects reads from either the SOLiD or Solexa sequencer. PerM guesses the read file format according to the first read file's ext name, ex: ".fasta", ".fastq", ".csfasta", ".csfastq". User can also use arg "--readFormat [csfata, fasta, fastq]> to specify the reads files format, if the input file has a different ext name. The read length are detected, according to the file format. Reads file format and the read lengths are assumed to be equal to the first read in the reads file list (.txt file). 

==== SOLiD reads' quality ====
For SOLiD reads in csfasta format, PerM will try to read the file with the same basename as the read set, however with the ext name ".qual" for quality score. That is, if the read file name is 7B.csfasta, its quality file "7B.qual" should be put beside.
If quality score file is not available, or the reads and quality scores can't be matched, only the read sequences will be used. 
(Tips! Use our shell splitReads.sh to split fasta or csfasta file into one million reads per set. It will generate a list file. Use the list file name as the input ptr of PerM.)

==== Paired-end reads ====
PerM takes two format of the paired end reads. 
(1) Concatenated 5'-3' end 3'-5' fasta file. 
(2) Two separated file of with _F3 and _R3. Each corresponding line is paired.
For input format (1), each file name should be listed in a row as single-end read file. 
For input format (2), the two paired files can be specified in a list, with each row having two files path, separated by tab. Ex: reads_F3.csfasta \t reads_R3.csfasta. 
Or, specified with the command line for a paired read files:

{{{./PerM <reference or index path> -1 <forward reads path> -2 <reverse reads path> [-flag options] }}}

[http://code.google.com/p/perm/wiki/Manual Back to manual]