==Mapping Summary==
The summary of the mapping results are printed from stdout. They include the number of reads mapped with each number of mismatches and the running time of each step. 
Use " | tee mapping.log" to get the summary and "grep" for the statistics. 
Ex:
{{{ cat stats.txt | tr '\r' '\n' | grep 'Sub' }}}

{{{ cat stats.txt | tr '\r' '\n' | grep 'Read#' }}}

You can also use the script 
[http://perm.googlecode.com/files/PerMStat.sh PerMStat.sh] to get mapping statistics for multiple mapping read sets.

==Mapping Results==
=== SAM Format ===
PerM can output in *SAM* format, by specify the flag -o `*`.sam.
Users can use the samtools to do some downstream analysis, ex: pileup read to get the consensus seq. For more information about sam format, check http://samtools.sourceforge.net/ and its [http://sourceforge.net/apps/mediawiki/samtools/index.php?title=SAM_protocol#C_APIs_.28SAMtools-C.29 wikipage]. 
Once the reads is pileup, you can also 

{{{samtools pileup -S -f <reference fasta> <sam file> | awk '{print $1, $2-1, $2, $4}' > mywiggle.bed }}}

 use the command to get the wig file with base signal. Further process can make a condensed wig for display.
 
The GUI tool [http://bioinf.scri.ac.uk/tablet/ Tablet] can be used to visualize the *sorted* sam output. You can also try [http://people.virginia.edu/~arq5x/utilities.html Araon's Quinlan]'s [http://people.virginia.edu/~arq5x/files/utilities/python/samToBed.py python script] to translate the sam format to bed file to show result in the genome browser. 

Here I described the the original output format, which has the ext name `*`.mapping. 
=== PerM's Mapping Format ===
====Single-end read====
Each line shows a qualified mapping info in the following order, separated by tab:
  * Read Id (Extract from the previous line of the reads input file. If the Id is missing, a fake tag will be used.)
  * The read sequence. For SOLiD reads, the first base from the agent (not in the reference) is eliminated from display. The first color signal is translated into a base for display.
  * Chromosome Name or ID (begins at 0 and numbered according to the order in the reference list)
  * Chromosome Locus ID, the most left position on the reference. (Each chromosome has indexes starts from 0). 
  * The mapped reference sequence. (reversed-complemented if orientation is -). For SOLiD reads, this region is translated into colors for comparison to the read.
  * Positive (forward) or negative (backward) strand that the reads mapped to.
  * Number of mismatches or the mismatches score (the sum of quality scores in the mis-matched bases) if available. 
  * Number of mappings that this read aligns to, including the current one. 
  * Reads qualities, numbers printed in characters with a Illimina Quality score shift. 

The last column for the quality scores will be missed if the input read is in fasta format (no quality scores). To grep the uniquely mapped read, use 
{{{grep -P '\t1\t[!\t]*\t$' [result].mapping }}} to extract the line with last second column with value 1. 
 
For Illumina reads with quality scores(fastq file), users can use the tool [http://evolution.sysu.edu.cn/mapview/ MapView] to visualize the single-end result with the [http://perm.googlecode.com/files/MVFmaker_NewFormat.txt configure file] on Windows. 

====Paired-end read====
  * Read Id (Extract from the previous line of the forward reads input file. If the Id is missing, a fake tag will be used.)
  * The 1st read sequence. 5' to 3'.
  * The 2nd read sequence. 5' to 3'.
  * The 1st read mismatches score.
  * The 2nd read mismatches score.
  * The ref name of the the 1st read mapped to.
  * The ref name of the the 2nd read mapped to.
  * The position of the the 1st read mapped to.
  * The position of the the 2nd read mapped to.
  * Category, this is the flag defined by ABI, which is a three letter alphabet. 
  see [http://www3.appliedbiosystems.com/cms/groups/mcb_marketing/documents/generaldocuments/cms_058718.pdf Here 2.2.9 c] to get detail info. 

==Unmapped reads==
With arg flag -u specified, PerM output the unmapped reads. The format is in fasta or fastq, depending on whether the quality score is available. I need to add the .csfasta format in the near future. Note the paired-end mapping result of the sam format includes the mapped single-end. 

== Index ==
From version 0.1.8, PerM's index has changed to include reference sequence in binary format. Users can specify the index path, however must keep the ext name as .index
The default index name has a format {{{<ReadLength>_<Seed>_<ReadType>_<RefFileName>_<Version>.index}}}

[http://code.google.com/p/perm/wiki/Manual Back to manual]