#labels Featured
=Output=
The summary of the mapping results are printed directly to the screen. They include the  number of reads mapped with each number of mismatches and the running time of each step. 
use " | tee mapping.log" to get the summary and "grep" for the statistics. 

PerM outputs the mapping summary stats in the standard output. Here I show the example of bash command to parse the stats. 
{{{ PerM ref.fa reads.fa -v 3 | tee stats.txt}}}
{{{ cat stats.txt | tr '\r' '\n' | grep 'Sub' }}}
{{{ cat stats.txt | tr '\r' '\n' | grep 'Read#' }}}

PerM generates (1)index (2) mapping result (3) Unmapped reads files.
== Index ==
From version 0.1.8, PerM's index has changed to include reference sequence in binary format. User can specify the index path, however must keep the ext name as .index
The default index name has a format {{{<ReadLength>_<Seed>_<ReadType>_<RefFileName>_<Version>.index}}}

==Mapping Results==
PerM can output in SAM format, by specify the flag -o `*`.sam.
User can use the using samtools to do some down stream analysis, ex: pileup or consensus read. For more information about sam format, check http://samtools.sourceforge.net/

Here I described the the original output format, which has the ext name `*`.mapping. 
===Single-end read===
Each line shows a qualified mapping info in the following order, separated by tab:
  * Read Tag (Extract from the previous line of the reads input file. If the tag is missing, a fake tag will be used.)  
  * The read sequence. For SOLiD reads, the first base from the agent (not in the reference) is eliminated from display. The first color signal is translated into a base for display.
  * Chromosome Name or ID (begins at 0 and numbered according to the order in the reference list)
  * Chromosome Locus ID, the most left position on the reference. (Each chromosome has indexes starts from 0). 
  * The mapped reference sequence. (reversed-complemented if orientation is -). For SOLiD reads, this region is translated into colors for comparison to the read.
  * Positive (forward) or negative (backward) strand that the reads mapped to.
  * Number of mismatches or the mismatches score (the sum of quality scores in the mis-matched bases) if available. 
  * Number of mappings that this read aligns to, including the current one. 
  * Reads qualities, numbers printed in characters with a Illimina Quality score shift 
===Paired-end read===

==Unmapped reads==
With arg flag -u specified, PerM output the unmapped reads. The format is in fasta or fastq, depending on whether the quality score is available. I need to add the .csfasta format in the near future. Note the paired-end mapping result of the sam format includes the mapped single-end. 

[http://code.google.com/p/perm/wiki/Usage Back to usage]