= Introduction =
PerM uses both configure file and simple command line options, which should be easy enough to make test. We are still working hard to standardize our I/O and options with other mapper, especially with Applied Biosystems pipeline.
= The command line options =
Given no command line option, PerM will try to open a default configure file name "config.pref". Given only one command line option, it should be the name of the configure file. The configure file format is described in the next section.

The currently command line option requires input in orders. It needs 
  * A reference .fasta file or a .txt which has many fasta file names, each in a row. 
  * A sequence file with .fasta, .prb or.csfasta formats or a .txt file as the list of those files.
  * Automatic links to other wiki pages


requires three input files - a FASTA-formatted reference genome file, and the two files generated by the SOLiD system that contain raw color data and quality scores (xxx.csfasta and xxx_QV.qual, respectively). The program produces a list of best matches for mapped reads and six genome-wide sets of coverage scores as its output. The first two coverage maps correspond to the raw coverage scores for all sequence reads that can be mapped unambiguously (one for each strand), and a separate pair contains coverage scores for all reads that match multiple locations within the reference genome. The final pair of files contains a record of valid sequence-space variants encountered between the sequence data set and the reference genome (both frequency and identity).


* Text in *bold* or _italic_