#labels Featured
If you have any usage question, please email "yanghoch at usc.edu".

== CONTENTS ==  
=== [http://code.google.com/p/perm/wiki/Compile Compile] ===
=== [http://code.google.com/p/perm/wiki/Input Input Files]===
=== [http://code.google.com/p/perm/wiki/Output Output Files]===
=== [http://code.google.com/p/perm/wiki/QAin2008USC_MCB_Retreat Q & A ] ===
=== [http://code.google.com/p/perm/wiki/ProgramComparisons Comparison ] ===
=== [http://code.google.com/p/perm/wiki/Limitation Limitation and Future work] ===
== SYNOPSIS ==  
To use the command line, type PerM with the args in the order.

For single end read: {{{PerM <Ref> <Reads> [options]}}}

For example:
 * {{{./PerM Ref.fasta Reads.fasta -v 5 }}}
 * {{{./PerM RefFilesList.txt ReadsSetFilesList.txt -v 5 -u unmappedReads.fa -E }}}
 * {{{./PerM Ref.fasta Reads.csfasta -v 5 -m -s my.index --delimiter ',' --seed F3}}}
 * {{{./PerM my.index SingleEndReads.csfasta -v 5}}} 
For paired end read: {{{ PerM <Ref> -1 <F3_Reads> -2 <R3_Reads> [options] }}}

For example:
 * {{{./PerM ref.fa -1 F3.fa -2 R3.fa -U 3000 -L 100 -v 5 -A -m -s -o out.sam}}}
 * {{{./PerM ref.txt -1 F3.fq -2 R3.fq -v 5 -m -s my.index -o out.mapping --seed F3}}}
 * {{{./PerM my.index -1 F3.fq -2 R3.fq -U 3000 -L 100 -v 5 -A -o out.sam}}}
== Command Line Options ==  
The following are the short description for each arguments:
 # The reference file should be in fasta format with the corresponding ext name ex: .fasta, .fna. or .fa. For transcriptom with multiple genes or isoforms as reference, concatenate all fasta in a fasta file. Or, if there are many files, one for each chromosome, ex: chr1.fa to chrY.fa, list them in a .txt file, so PerM knows it is a file listing the path of multiple fasta file.  
 # The reads file should be in the .fasta, .fastq, .csfasta or .csfastq format. PerM parses a file according to it's ext file name, or the format specified by flag *{{{--format <format>}}}*. If there are multiple read files, list each file names in a .txt file, with each row a path of the reads file. PerM takes it as the input and can map multiple read sets in parallel by OpenMP. 
 # Mismatch threshold could be a integer 2 to 5. For reads over 64, PerM allows double the number of mismatches than the given threshold.
 # Options: By default, PerM outputs all the best alignments in terms of the mismatches number with in the threshold, end-to-end. It will try to use the saved indexes. If not found, reconstruct the indexes. The following option flags can change the default settings.
 * *-A*     Output all alignments within the mismatch threshold.
 * *-B*     Output best alignments in terms of mismatches in the threshold.(Default) 
 * *-E*     Output only the unique mapped reads. 
 * *-m*     Create the reference index, without reusing the saved index.
 * *-s P*     Save the reference index to accelerate the mapping in the future. If path P is not specify, the default path will be used. 
 * *-v I*  where I i the number of mismatches allowed in one end.
 * *-T I*, where I is tbe length to truncate read length, say 30 means use only first 30 
bases (signals). Don't put the number if the full read is meant to be used.
 * *-o P*, where P is the the output path for one read set. PerM's output are in .mapping or .sam format, determined by the ext name of P. 
 * *-d P*, where P is the the output dir for multip read set
 * *-u P*, Print the fasta(fastq) file of those unmapped reads to a file
 * *--ignoreQS* Ignore the quality scores in fastq or QUAL files.
 * *--seed {F2 | S11 | F3 | F4}* Use the specified seed pattern.
 * *--readFormat {fasta | fastq | csfasta | csfastq}*. Read in reads in the specified format, instead of gussing according to the ext name.
 * *--delimiter c*, where c is a character which is used as the delimiter which is used to parse the read id. This is used for read set which attach additional information in the read id, using character ',', ':', '|' or other symbols. 

Ex:  ./PerM Ref.fa Reads.fa -v 5 -E -T 36 -m -s -o out.sam
PerM will report only the uniquely mapped reads within 5 mismatches in the first 36 bases.
PerM will reconstruct and save the indexes to the default path for future mappings.
PerM will output the mappings file to out.sam in the sam format.

====Include or Exclude Ambiguous Mapping====
The default setting will print the best alignments for each read in term of mismatches number. Use "-E" to exclude the reads has multiple "best" alignments. Use "-A" flag if all alignments within substitution threshold are meant to be printed. A maximum 100 mapping per read will be print consecutively in the output file. 

==Index==
After version 0.1.8, PerM save references sequence in binary in the index. It requires 4.5 bytes (one integer + 4 bits) + 256M memory for mapping. As always, the memory and disk should have enough capacity. 

= Paralleled Mapping =
PerM simultaneously maps multiple reads sets in a list by querying the same index. It will detect how many CPU (cores) are available and assign each of them a read set. If a read set is done, the next read set in the list will be processed automatically. Each read set will have its own output file `*`.mapping. To better utilize all CPUs on a node, large reads are suggested to be split to many small read sets and put in the list. 
When multiple nodes are used in the same file system, the index should be pre-built first by one node; the other node will read the pre-built index with out building index again. Without pre-built index, each machine will try to build their own index, wasting CPU hours and may cause insufficiency of disk quota. 

= Mate-paired Reads =
PerM deals with mate-paired reads by mapping each end separately. All combination of mated paired mapping in the same ref will be printed. Note this makes the running time become very slow, if both ends are in the repetitive regions. 

PerM takes two format of the paired end reads. 
(1) Concatenated 5'-3' end 3'-5' fasta file. 
(2) Two separated file of with _F3 and _R3. Each corresponding line is paired.
For input format (1), each file name should be listed in a row as single-end read file. 
For input format (2), the two paired files can be specified in a list, with each row having two files path, separated by . Ex: reads_F3.csfasta \t reads_R3.csfasta. 
Or, specified with the command line for a paired read files:
./PerM <reference or index path> -1 <forward reads path> -2 <reverse reads path> [-flag options]
(4)  Options for mate-paired reads:
-e     Exclude ambiguous paired. 
-L I   mate-paired separate lower bound
-U I   mate-paired separate upper bound, 
-o P   where P is the path (single paired read set) or the directory (multiple paired read sets)
The default is set as -L 0 -U 3000
The upper bound and lower bound can be negative, which may catch the re-arangement variations.
Ex:  ./PerM -1 F3.fq -2 R3.fq -U 500 -L 200 -m -s -o out.sam.
PerM will report only the non-ambiguous mate pair in the range of [200bp, 500bp]
PerM will reconstruct and save the indexes to the default path for future mappings.
PerM will output the mappings file to out.sam in the sam format.