#labels Featured
= Options =
PerM uses both configuration file or command line options. The input files and the configuration file should be put beside the program.

== The command line options ==  
The currently command line options are required to be in the order as follows:
  # A reference in .fasta file or a .txt which has many fasta file names, each in a row. 
  # The reads file in .fasta, .prb or .csfasta formats or a .txt file as the list of those files.
  # The full sensitivity threshold.

  * Ex: {{{./PerM_P anthracis.fasta 7B.csfasta 3}}}
  * Ex: {{{./PerM_P HumanGenome.txt IlluminaReadSim2.txt 2}}}
====Reference====
Be careful about the file name length of the references in the list. Because PerM concatenates all input reference file names to generate the output file name, the output file name could be longer then the OS limitation, say 256 characters. If this happens, try to rename the length of reference file names to make them shorter. 

====Reads file====
PerM expects reads are all from SOLiD or all from Solexa sequencer. If the first read file has the file name extension ".csfasta", PerM expects all reads are from SOLiD. PerM expects all reads have the same length, which is detected automatically using the first read in the first read file by PerM.
PerM uses *parallel for* in openMP, which assigns each core/CPU a read set simultaneously . Once a read set is done, another read set in the list will be assigned to the available CPU. To fully utilize your computing power, large read set can be divided into many small read sets manually.

====Full sensitivity threshold====
PerM support normal full sensitivity including 2, 3, 4 mismatches. For the special threshold of SOLiD, "one color + one base", *25* is used as the input parameter. For the special threshold of SOLiD, "Two base", *33* is used as the input parameter.   
Ex: {{{./PerM_P anthracis.fasta 7B.csfasta 25}}}

== The configuration file ==  
Given no command line option, PerM will try to open a default configuration file name "config.pref". Given only one command line option, it should be the name of the configuration file. The follow is the examples, as an attached with the code tar-ball :
{{{
anthracis.fasta
7B.csfasta
#DummyQualitySCORE
output.txt
25
10000
1
false
1
0
}}}
The first two rows are the *Reference* and *Read file*. The fifth row shows the full sensitive threshold. The rest parameters are used by the preprocessed reads version only. 

=OUTPUT=
==Index==
==Mapping Results==
====Solexa Reads ====
====SOLiD Reads ====

Changes are expected in the near future for I/O and options standardization with other mapper, especially Applied Biosystems pipeline.