#labels Featured
= Options =
PerM uses both configuration file and command line options. The input files and configuration files should be put beside the program.

== The command line options ==  
The command line options should be as follows:
  # A reference in .fasta file or a .txt which has many fasta file names, each in a row. 
  # The reads file in .fasta, .prb or .csfasta formats or a .txt file as the list of those files.
  # The full sensitivity threshold.

  * Ex: {{{./PerM_P anthracis.fasta 7B.csfasta 3}}}
  * Ex: {{{./PerM_P HumanGenome.txt IlluminaReadSim2.txt 2}}}
====Reference====
Because PerM concatenates all input file names to generate the output file name, the output file name could be longer then the OS limit, say 256 characters. If this happens, rename and limit the length of reference files.

====Reads file====
PerM expects reads are all from SOLiD or all from Solexa sequencers. If the first read file has the file name extension ".csfasta", PerM expects all reads are from SOLiD. PerM expects all reads have the same length as the first read in the first read set.

====Full sensitivity threshold====
PerM support normal full sensitivity including 2, 3, 4 mismatches. For the special threshold of SOLiD, "one color + one base", *25* is used as the input parameter. For the special threshold of SOLiD, "Two base", *33* is used as the input parameter.   
Ex: {{{./PerM_P anthracis.fasta 7B.csfasta 25}}}

== The configuration file ==  
Given no command line option, PerM will try to open a default configuration file name "config.pref". Given only one command line option, it should be the name of the configuration file. The follow is the examples, as an attached with the code tar-ball :
{{{
anthracis.fasta
7B.csfasta
#DummyQualitySCORE
output.txt
25
10000
1
false
1
0
}}}
The first two rows are the *Reference* and *Read file*. The fifth row shows the full sensitive threshold. The rest parameters are used by the preprocessed reads version only. 

=OUTPUT=
The summary of the mapping are printed out on the screen directly, including the number of reads mapped within the number of mismatches and the running time of each steps.
==Index==
The genome-preprocessed version will save the index of the reference by default. It needs 4.5 bytes (one integer + 4 bits) + 256M memory to do the mapping. Make sure the memory and disk has enough capacity.
==Mapping Results==
The result of mapping are printed in a `*`.mapping file with name corresponding to the reference and reads. By default, the output includes non-ambiguous reads only. Each lines is the result of a best mapping in terms of number of mismatches. The info of each line is in the following orders:
  * ReadID (the order of the read in the read set)
  * The reads. For SOLiD reads, the first base from the agent (not in the reference) is eliminated from display. The first color signal is translated into a base for display.
  * Chromosome ID (start from 0 and is numbered according to the order in the reference list.
  * Chromosome Locus ID (start from 0 and the first position of each chromosome. 
  * The mapped reference region. For SOLiD reads, this region is translated into colors to compared with reads.
  * Positive (forward) or negative (backward) strand that the reads mapped to.
  * Number of mismatches in the mapping. 


Changes are expected in the near future for I/O and options standardization with other mapper, especially Applied Biosystems pipeline.