= Options =
PerM uses both configuration file and command line options. The input files and configuration files should be put besides the program.

== The command line options ==  
The currently command line option requires input in order as follows:
  # A reference in .fasta file or a .txt which has many fasta file names, each in a row. 
  # The reads file in .fasta, .prb or .csfasta formats or a .txt file as the list of those files.
  # The full sensitivity threshold.

  * Ex: {{{./PerM_P anthracis.fasta 7B.csfasta 3}}}
  * Ex: {{{./PerM_P HumanGenome.txt IlluminaReadSim2.txt 2}}}
=== Reference ===
Be careful about the number of references in the list. Because PerM concatenates all input file names to generate the output file name, the output file name could be longer then the O/S limit, say 256 characters. If this happens, try to rename and limit the length of reference files' names. 

=== Reads file ===
PerM expects reads are all from SOLiD or all from Solexa sequencer. If the first read file has the filename extension ".csfasta", PerM expects all reads are from SOLiD. PerM expects all reads have the same length, which is detected automatically using the first read in the first read file by PerM.

=== Full sensitivity threshold ===
PerM support normal full sensitivity including 2, 3, 4 mismatches. For the special threshold of SOLiD, "one color + one base", *25* is used as the input parameter. For the special threshold of SOLiD, "Two base", *33* is used as the input parameter.   
Ex: {{{./PerM_P anthracis.fasta 7B.csfasta 25}}}

== The configuration file ==  
Given no command line option, PerM will try to open a default configuration file name "config.pref". Given only one command line option, it should be the name of the configuration file. The format is described as the examples:
{{{
}}}

= Test Data =
=== Reference ===
  #The anthracis.fasta can be download in SOCS webpage: [http://socs.biology.gatech.edu/Download.html] 
  #The repeated masked and non-masked human genome can be downloaded from UCSC's genome browser website with file chromFaMasked.zip and chromFa.zip respectively:
  [http://hgdownload.cse.ucsc.edu/goldenPath/hg18/bigZips]. \*.fa files are rename the *.fa file to *.fasta file manually. 
=== Reads file ===
We used real SOLiD data down from SOCS page and the simulated data gen

Changes are expected in the near future for I/O and options standardization with other mapper, especially Applied Biosystems pipeline.